E0225 needs to be updated to new format

From: src/test/compile-fail/E0225.rs

Error E0225 needs a span_label, updating it from:

```
error[E0225]: only the builtin traits can be used as closure or object bounds
  --> src/test/compile-fail/E0225.rs:12:32
   |
12 |     let _: Box<std::io::Read + std::io::Write>; //~ ERROR E0225
   |                                ^^^^^^^^^^^^^^
```

To:

```
error[E0225]: only the builtin traits can be used as closure or object bounds
  --> src/test/compile-fail/E0225.rs:12:32
   |
12 |     let _: Box<std::io::Read + std::io::Write>; //~ ERROR E0225
   |                                ^^^^^^^^^^^^^^ non-builtin trait used as bounds
```

--- a/src/librustc_typeck/astconv.rs
+++ b/src/librustc_typeck/astconv.rs
@@ -2091,8 +2091,11 @@ impl<'o, 'gcx: 'tcx, 'tcx> AstConv<'gcx, 'tcx>+'o {
 
         if !trait_bounds.is_empty() {
             let b = &trait_bounds[0];
-            span_err!(self.tcx().sess, b.trait_ref.path.span, E0225,
-                      "only the builtin traits can be used as closure or object bounds");
+            let span = b.trait_ref.path.span;
+            struct_span_err!(self.tcx().sess, span, E0225,
+                             "only the builtin traits can be used as closure or object bounds")
+                .span_label(span, &format!("non-builtin trait used as bounds"))
+                .emit();
         }
 
         let region_bound =

--- a/src/test/compile-fail/E0225.rs
+++ b/src/test/compile-fail/E0225.rs
@@ -9,5 +9,7 @@
 // except according to those terms.
 
 fn main() {
-    let _: Box<std::io::Read + std::io::Write>; //~ ERROR E0225
+    let _: Box<std::io::Read + std::io::Write>;
+    //~^ ERROR only the builtin traits can be used as closure or object bounds [E0225]
+    //~| NOTE non-builtin trait used as bounds
 }

